src/e84_geoai_common/llm/core/llm.py

```diff
@@ -54,30 +85,32 @@ def to_text_only(self) -> str:
         return " ".join(parts)
 
 
-class Tool(BaseModel):
-    """Defines a tool that the model may use."""
+class LLMTool(BaseModel):
+    """Definition of a tool that an LLM may use."""
 
     model_config = ConfigDict(strict=True, extra="forbid")
 
     name: str
     description: str
-    input_schema: dict[str, Any]
-
-    @classmethod
-    def from_function(cls, func: Callable[..., Any]) -> Self:
-        """Construct from a Python funtion."""
-        # This works because our tool class has the same fields as a Claude tool
-        schema = get_function_schema(func, format="claude")  # type: ignore[reportUnknownVariableType]
-        return cls.model_validate(schema)
+    input_model: type[BaseModel] | None = Field(
+        description="A Pydantic model describing the inputs to the tool. "
+        "Can be set to None to indicate that the tool takes no inputs."
+    )
+    output_model: type[BaseModel] | None = Field(
+        description="A Pydantic model describing the output of the tool. "
+        "Can be set to None to indicate that the tool returns no text or JSON outputs."
+    )
 
 
-class ToolUse(BaseModel):
-    """Identifies a selected tool to use."""
+class LLMToolChoice(BaseModel):
+    """Specification for constraining an LLM's choice of tools."""
 
     model_config = ConfigDict(strict=True, extra="forbid")
 
-    name: str
-    input: dict[str, Any]
+    mode: Literal["optional", "force_tool_use", "force_specific_tool_use"] = "optional"
```

Comment:
Can you add a comment here describing what these options are for? (I think I understand it but others might not)